src/core/packages/chrome/browser/src/project_navigation.ts

```diff
@@ -194,6 +194,11 @@ interface NodeDefinitionBase {
    * -------------------------------- ITEM NODES ONLY PROPS ---------------------------------------
    * ----------------------------------------------------------------------------------------------
    */
+  /**
+   * Handler to render the node item with custom JSX. This handler is added to render the `children` of
+   * the Navigation.Item component when React components are used to declare the navigation tree.
+   */
+  renderItem?: () => React.ReactNode;
```

Comment:
If this is needed in the `NodeDefinitionBase` interface, it can be removed from the `ChromeProjectNavigationNode` interface